core/src/main/java/nl/nn/adapterframework/jms/BytesMessageInputStream.java

```diff
@@ -32,14 +33,19 @@ public BytesMessageInputStream(BytesMessage bytesMsg) {
 	@Override
 	public int read() throws IOException {
 		try {
-			return bytesMsg.readByte();
+			byte[] data = new byte[1];
+			int len = bytesMsg.readBytes(data);
+			if (len == -1) {
+				return -1;
+			}
+			return data[0] & 0xFF;
```

Comment:
Why this interesting bit flipping? Can you add a comment on the purpose of this magic?

Reply:
This was suggested by IntelliJ to ensure the return data was unsigned. You are absolutely right this needs a comment with explanation.